coriolis/tests/conductor/rpc/test_server.py

```diff
@@ -1556,3 +1556,282 @@ def create_task_side_effect(
+    @mock.patch.object(uuid, "uuid4")
+    @mock.patch.object(models, "TasksExecution")
+    @mock.patch.object(
+        server.ConductorServerEndpoint,
+        "_check_create_reservation_for_transfer"
+    )
+    @mock.patch.object(
+        server.ConductorServerEndpoint,
+        "_check_minion_pools_for_action"
+    )
+    @mock.patch.object(
+        server.ConductorServerEndpoint,
+        "_get_instance_scripts"
+    )
+    @mock.patch.object(
+        server.ConductorServerEndpoint,
+        "_create_task"
+    )
+    @mock.patch.object(
+        server.ConductorServerEndpoint,
+        "_check_execution_tasks_sanity"
+    )
+    @mock.patch.object(
+        db_api,
+        "add_migration"
+    )
+    @mock.patch.object(
+        lockutils,
+        "lock"
+    )
+    @mock.patch.object(
+        server.ConductorServerEndpoint,
+        "_minion_manager_client"
+    )
+    @mock.patch.object(
+        server.ConductorServerEndpoint,
+        "_set_tasks_execution_status"
+    )
+    @mock.patch.object(
+        server.ConductorServerEndpoint,
+        "_begin_tasks"
+    )
+    @mock.patch.object(
+        server.ConductorServerEndpoint,
+        "get_migration"
+    )
+    @ddt.file_data("data/migrate_instances_config.yml")
+    @ddt.unpack
+    def test_migrate_instances(
+            self,
+            mock_get_migration,
+            mock_begin_tasks,
+            mock_set_tasks_execution_status,
+            mock_minion_manager_client,
+            mock_lock,
+            mock_add_migration,
+            mock_check_execution_tasks_sanity,
+            mock_create_task,
+            mock_get_instance_scripts,
+            mock_check_minion_pools_for_action,
+            mock_check_create_reservation_for_transfer,
+            mock_tasks_execution,
+            mock_uuid4,  # pylint: disable=unused-argument
+            mock_migration,
+            mock_get_provider_types,
+            mock_check_endpoints,
+            mock_get_endpoint,
+            config,
+            expected_tasks,
+    ):
+        has_origin_minion_pool = config.get(
+            'has_origin_minion_pool', False
+        )
+        has_destination_minion_pool = config.get(
+            'has_destination_minion_pool', False
+        )
+        shutdown_instances = config.get('shutdown_instances', False)
+        skip_os_morphing = config.get('skip_os_morphing', True)
+        get_optimal_flavor = config.get('get_optimal_flavor', False)
+
+        if get_optimal_flavor:
+            mock_get_provider_types.return_value = [
+                constants.PROVIDER_TYPE_INSTANCE_FLAVOR
+            ]
+
+        instances = [
+            mock.sentinel.instance_1,
+            mock.sentinel.instance_2,
+        ]
+        instance_osmorphing_minion_pool_mappings = {}
+        # use morphing pool mappings only
+        # if origin and destination minion pools are used,
+        # this way we can test the case when we don't use any minion pools
```

Comment:
Why wouldn't/couldn't we also test osmorphing pools without having origin/destination pools? You can configure this separately in your yml file, add a `has_osmorphing_pool`, and check that instead of either the origin or destination pools

Reply:
I initially considered skipping tests specific to the osmorphing pool for some reason. However, I have now updated the test to address this issue.